idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1619,4 +1619,16 @@ struct SupportedClientVersions{
 // ClusterInfo contains information about cadence cluster
 struct ClusterInfo{
   10: optional SupportedClientVersions supportedClientVersions
+}
+
+struct GetRawHistoryRequest {
+  10: optional string domain
+  20: optional WorkflowExecution execution
+  30: optional i32 maximumPageSize
+  40: optional binary nextPageToken
+}
+
+struct GetRawHistoryResponse {
+  10: optional list<binary> rawHistory
```

Comment:
If you only return a binary, the client will have no idea how to decode the binary. You also need to return some metadata information. Please check the `DataBlob` struct.